nbstripout/_utils.py

```diff
@@ -153,3 +157,106 @@ def strip_output(nb, keep_output, keep_count, extra_keys=[], drop_empty_cells=Fa
         for field in keys['cell']:
             pop_recursive(cell, field)
     return nb
+
+
+def process_pyproject_toml(toml_file_path: str, parser: ArgumentParser) -> Optional[Dict[str, Any]]:
+    """Extract config mapping from pyproject.toml file."""
+    try:
+        import tomllib  # python 3.11+
+    except ModuleNotFoundError:
+        import tomli as tomllib
+
+    with open(toml_file_path, 'rb') as f:
+        dict_config = tomllib.load(f).get('tool', {}).get('nbstripout', None)
+
+    if not dict_config:
+        # could be {} if 'tool' not found, or None if 'nbstripout' not found
+        return dict_config
+
+    # special processing of boolean options, make sure we don't have invalid types
+    for a in parser._actions:
+        if a.dest in dict_config and isinstance(a, (_StoreTrueAction, _StoreFalseAction)):
```

Comment:
Yes, I thought that this would still be more robust than keeping an explicit list of Boolean arguments, although that wouldn't be impossible. I looked at a range of options, including shifting to an alternative arguments library, like Click, but that all got too complicated.
I think that this is a reasonable compromise if we keep on top of adding new Python versions to the test suite whenever they become available.

Reply:
Yep, agree